packages/edit-site/src/components/sidebar-edit-mode/page-panels/page-status.js

```diff
@@ -0,0 +1,229 @@
+				newDate = null;
+			}
+		} else if ( value === 'future' ) {
+			if ( ! date || new Date( date ) < new Date() ) {
+				newDate = new Date();
+				newDate.setDate( newDate.getDate() + 7 );
+			}
+		}
+		saveStatus( {
+			status: value,
+			date: newDate,
+		} );
+	};
+
+	return (
+		<HStack className="edit-site-summary-field">
+			<Text className="edit-site-summary-field__label">
+				{ __( 'Status' ) }
+			</Text>
+			<Dropdown
+				contentClassName="edit-site-change-status__content"
+				popoverProps={ popoverProps }
+				focusOnMount
+				ref={ setPopoverAnchor }
+				renderToggle={ ( { onToggle } ) => (
+					<Button
+						className="edit-site-summary-field__trigger"
+						variant="tertiary"
+						onClick={ onToggle }
+					>
+						<StatusLabel
+							status={ password ? 'protected' : status }
+						/>
+					</Button>
+				) }
+				renderContent={ ( { onClose } ) => (
+					<>
+						<InspectorPopoverHeader
+							title={ __( 'Status' ) }
+							onClose={ onClose }
+						/>
+						<VStack spacing={ 5 }>
+							<BaseControl
+								hideLabelFromVision
+								label={ __( 'Status' ) }
+								id={ `edit-site-change-status__status` }
+								className={ 'components-radio-control' }
+							>
+								<VStack spacing={ 3 }>
+									{ STATUS_OPTIONS.map( ( option, index ) => (
+										<div
+											key={ `edit-site-change-status__status-${ index }` }
+											className="components-radio-control__option with-hint"
+										>
+											<input
+												id={ `edit-site-change-status__status-${ index }` }
+												className="components-radio-control__input"
+												type="radio"
+												name={ `edit-site-change-status__status` }
+												value={ option.value }
+												onChange={ ( e ) =>
+													handleStatus(
+														e.target.value
+													)
+												}
+												checked={
+													option.value === status
+												}
+											/>
+											<VStack spacing={ 1 }>
+												<label
+													htmlFor={ `edit-site-change-status__status-${ index }` }
+												>
+													{ option.label }
+												</label>
+												{ option.hint && (
+													<Text variant="muted">
+														{ option.hint }
+													</Text>
+												) }
+											</VStack>
+										</div>
+									) ) }
+								</VStack>
+							</BaseControl>
+							<BaseControl
+								id={ `edit-site-change-status__password` }
+								label={ __( 'Password' ) }
+							>
+								<ToggleControl
+									label={ __(
+										'Hide this page behind a password'
+									) }
+									checked={ showPassword }
+									onChange={ handleTogglePassword }
+								/>
+								{ showPassword && (
+									<TextControl
+										onChange={ ( value ) =>
+											saveStatus( { password: value } )
+										}
+										value={ password }
+										/* eslint-disable jsx-a11y/no-autofocus */
+										autoFocus
+										/* eslint-enable jsx-a11y/no-autofocus */
```

Comment:
This also means that if you've set a password and closed the popover, when you reopen it again, the focus will be moved to here instead of the first focusable element. This is probably not what we want?

Reply:
Good point. I've updated to only focus input when password is empty.